packages/server-wallet/src/models/adjudicator-status.ts

```diff
@@ -0,0 +1,88 @@
+import {getChannelMode} from '@statechannels/nitro-protocol';
+import {SignedState} from '@statechannels/wallet-core';
+import Knex from 'knex';
+import _ from 'lodash';
+import {Model} from 'objection';
+
+import {Bytes32, Uint48} from '../type-aliases';
+
+export type AdjudicatorStatus =
+  | {
+      channelMode: 'Finalized';
+      states: SignedState[];
+    }
+  | {channelMode: 'Challenge'; states: SignedState[]}
+  | {channelMode: 'Open'};
+
+interface RequiredColumns {
+  readonly channelId: Bytes32;
+  readonly finalizesAt: Uint48;
+  readonly blockNumber: Uint48;
+  readonly blockTimestamp: Uint48;
+}
+export class AdjudicatorStatusModel extends Model implements RequiredColumns {
+  readonly channelId!: Bytes32;
+  readonly finalizesAt!: Uint48;
+  readonly blockNumber!: Uint48;
+  readonly blockTimestamp!: Uint48;
+  readonly states!: SignedState[];
+  static tableName = 'adjudicator_status';
+  static get idColumn(): string[] {
+    return ['channelId'];
+  }
+
+  static async getAdjudicatorStatus(knex: Knex, channelId: Bytes32): Promise<AdjudicatorStatus> {
+    const result = await AdjudicatorStatusModel.query(knex).where({channelId}).first();
+
+    if (!result) return {channelMode: 'Open'};
+    return AdjudicatorStatusModel.convertResult(result);
+  }
+
+  static async setFinalized(
+    knex: Knex,
+    channelId: string,
+    blockNumber: number,
+    blockTimestamp: number
+  ): Promise<AdjudicatorStatus> {
+    const existing = await AdjudicatorStatusModel.query(knex).where({channelId});
+    if (!existing) {
+      await AdjudicatorStatusModel.query(knex).insert({channelId, blockNumber, blockTimestamp});
+    }
+    const result = await AdjudicatorStatusModel.query(knex)
+      .patch({blockNumber, blockTimestamp})
+      .where({channelId})
+      .returning('*')
+      .first();
+
+    return AdjudicatorStatusModel.convertResult(result);
+  }
+  static async insertAdjudicatorStatus(
+    knex: Knex,
+    channelId: string,
+    finalizesAt: number,
+    states: SignedState[]
+  ): Promise<AdjudicatorStatus> {
+    const result = await AdjudicatorStatusModel.query(knex).insert({
+      channelId,
+      finalizesAt,
+      blockNumber: 0,
```

Comment:
Curious, is there a reason we are choosing to make these columns non-nullable and defaulting to values of `0`? An alternative is to allow nulls.

Reply:
I think I did this for simplicity so that the `model` always returns a `number` when looking at the `blockNumber`\`blockTimestamp`  so we didn't have to add `model.BlockNumber ?? 0` in places.